model/user.go

```diff
@@ -47,6 +47,77 @@ const (
 	USER_PASSWORD_MAX_LENGTH  = 72
 )
 
+// This is a struct for a user interaction
+// A user interaction will represent one row in the results of the
+// GetUserInteractions query in store/sqlstore/user_store.go
+type UserInteraction struct {
+	// The type of interaction
+	// enum('Reaction', 'Mention', 'Reply', 'DirectMessage')
+	InteractionType string `json:"interaction_type"`
+
+	// The username of the user that this interaction took place with
+	Username string `json:"username"`
+
+	// Denotes whether or not the current user was the recipient of this interaction
+	IsRecipient bool `json:"is_recipient"`
+
+	// The context in which this interaction occurred (either 'Course' or the learning group type's prefix (ex. 'plg'))
+	Context string `json:"context"`
+
+	// The type of channel that this interaction took place in
+	// enum('P', 'O', 'D', 'G')
+	//     P = private channel
+	//     O = public channel
+	//     D = direct message (1 on 1)
+	//     G = group direct message
+	ChannelType string `json:"channel_type"`
+
+	// The slug name of the channel that this interaction took place in (Channels.Name)
+	ChannelName string `json:"channel_name"`
+}
+
+// This is a struct for a learning group type
+// Each learning group type is derived from personal channel names found in the config
+type LearningGroupType struct {
+	// ex. 'capstone'
+	Prefix string `json:"prefix"`
+}
+
+// This is a struct for all user analytics data sets
+// This is the struct that gets returned to mattermost-redux in the users/{userId}/teams/{teamId}/useranalytics request
+// **Note: Using pointers as property types allows for null values
+type UserAnalytics struct {
+	TargetUser         string               `json:"TargetUser"`
+	UserInteractions   []*UserInteraction   `json:"UserInteractions"`
+	UserLearningGroups []*LearningGroup     `json:"UserLearningGroups"`
+	LearningGroupTypes []*LearningGroupType `json:"LearningGroupTypes"`
+	// Add more datasets here
+}
+
+// This is a struct for a learning group (ex. capstone)
+// **Note: Using pointers as property types allows for null values
+type LearningGroup struct {
+	// The name of the learning group (the channel name minus the learning group type's prefix)
+	//     ex. plg-30 => 30
+	LearningGroupName string `json:"learning_group_name"`
+
+	// The type of learning group (personal channel name from the config)
+	//     ex. plg
+	LearningGroupType string `json:"learning_group_type"`
+
+	// The id of the private channel for this learning group
+	ChannelId string `json:"channel_id"`
+
+	// The slug name of the private channel for this learning group (Channels.Name)
+	ChannelName string `json:"channel_name"`
+
+	// A comma delimited list of the usernames of the members of this learning group (channel)
+	Members *string `json:"members"`
```

Comment:
I feel like there was a comment/discussion on this before, that Member should be a slice? But I don't see it.
It is [really easy](https://yourbasic.org/golang/split-string-into-slice/) to do in go and I think it makes sense to do it here rather than returning a string and having the recipient process it into an array later.
```
s := strings.Split("a,b,c", ",")
fmt.Println(s)
// Output: [a b c]
```
and define
```
Members []string `json:"members"`
```
I see that to actually accomplish this you'd need to transform the result of the query in the `store/sqlstore/user_store.go` to this format. I guess we can hold off on that for now because it's a little complicated, but I do think it would be better long term.